File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        result = []

        if not nums:
            return result
        
        start = nums[0]

        for i in range(1, len(nums)):
            if nums[i] != nums[i-1] + 1:
                end = nums[i-1]
                if start == end:
                    result.append(str(start))
                else:
                    result.append(f"{start}->{end}")
                start = nums[i]
        
        end = nums[-1]
        if start == end:
            result.append(str(start))
        else:
            result.append(f"{start}->{end}")
        
        return result



        
        

        

File: 0228-summary-ranges/0228-summary-ranges.dart
class Solution {
  List<String> summaryRanges(List<int> nums) {
    List<String> result = [];

    if (nums.isEmpty)
    {
        return result;
    }

    int start = nums[0]; 

    for (int i = 1; i < nums.length; i++)
    {
        if (nums[i] != (nums[i-1] + 1))
        {
            int end = nums[i-1];
            if (start == end)
            {
                result.add(start.toString());
            }
            else
            {
                result.add("$start->$end");
            }
            start = nums[i];
        }
    }

    int end = nums[nums.length - 1];
    if (start == end)
    {
        result.add(start.toString());
    }
    else 
    {
        result.add("$start->$end");
    }
    return result;

  }
}